### PR TITLE
Add helper functions to query properties of the lowered Target (#8192)

### PR DESCRIPTION
**Pull request by steven-johnson**

### CODE REVIEW

File: src/IROperator.h [hunk 1]
```diff
@@ -1689,6 +1690,45 @@ Expr rounding_mul_shift_right(Expr a, Expr b, Expr q);
...
+ * code without having to plumb a Target through call sites, so that you
+ * can do things like
+ \code
+    Expr e = select(target_arch_is(Target::ARM), something, something_else);
+ \endcode
+ */
+//@{
+Expr target_arch_is(Target::Arch arch);
+Expr target_os_is(Target::OS os);
+Expr target_has_feature(Target::Feature feat);
```
Comment (mcourteaux) [inline src/IROperator.h:1704]: This one should perhaps get a big warning that this does not check runtime-features of the CPU or device, but uses the compile-time target. I know it says that in the doc above, but I feel like many people will get fooled by their wrong intuition. It's especially tricky, as it returns an `Expr` and not a `bool`, so it gives off a runtime-vibe.

**Comment by steven-johnson** [comment]
This could be expanded to more of the Target API, but I stopped here; some other possibilities would be

```
bool target_supports_device_api(DeviceAPI device)
bool target_supports_type(const Type &t, DeviceAPI device)
```

**Review by abadams — APPROVED** [review]
This would address my use-case

**Comment by steven-johnson** [comment]
> Is there a way to get the device type of the enclosing loop for a given expression?

No, but it seems like it would be pretty simple to add, in the same manner of the helpers I just added. Feel free to open a PR to do so (it may be a little while before I can get to it).

**Pull request merged**